samza-core/src/test/java/org/apache/samza/serializers/model/serializers/TestIntermediateMessageSerde.java

```diff
@@ -126,4 +128,14 @@ public void testEndOfStreamMessageSerde() {
     assertEquals(de.getTaskName(), taskName);
     assertEquals(de.getVersion(), 1);
   }
+
+  @Test (expected = SamzaException.class)
+  public void testUserMessageSerdeException() {
```

Comment:
It seems like this test would have succeeded with the previous version of `IntermediateMessageSerde` as well. Could you please try to come up with a test that would only succeed with your new version of `IntermediateMessageSerde`?
For example, you could mock the `imserde2` and then verify that it only gets called with a certain byte array.

Reply:
Updated unit test to fail if the bytes given are not expected.